src/test/HostActivationTests/DotNetBuilder.cs

```diff
@@ -65,6 +65,67 @@ public DotNetBuilder AddMicrosoftNETCoreAppFramework(string version)
             return this;
         }
 
+        /// <summary>
+        /// Add a mock of the Microsoft.NETCore.App framework with the specified version
+        /// </summary>
+        /// <param name="version">Version to add</param>
+        /// <remarks>
+        /// Product runtime binaries are not added. All the added mock framework will contain is hostpolicy,
+        /// a mock version of coreclr, and a minimal Microsoft.NETCore.App.deps.json.
+        /// </remarks>
+        public DotNetBuilder AddMicrosoftNETCoreAppFrameworkMockCoreclr(string version)
+        {
+            // ./shared/Microsoft.NETCore.App/<version> - create a mock of the root framework
+            string netCoreAppPath = Path.Combine(_path, "shared", "Microsoft.NETCore.App", version);
+            Directory.CreateDirectory(netCoreAppPath);
+
+            // ./shared/Microsoft.NETCore.App/<version>/coreclr.dll - this is a mock, will not actually run CoreCLR
+            string hostPolicyFileName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("hostpolicy");
+            File.Copy(
+                Path.Combine(_repoDirectories.Artifacts, "corehost", hostPolicyFileName),
+                Path.Combine(netCoreAppPath, hostPolicyFileName),
+                true);
+
+            // ./shared/Microsoft.NETCore.App/<version>/coreclr.dll - this is a mock, will not actually run CoreCLR
+            string mockCoreclrFileName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("mockcoreclr");
+            File.Copy(
+                Path.Combine(_repoDirectories.Artifacts, "corehost_test", mockCoreclrFileName),
+                Path.Combine(netCoreAppPath, RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("coreclr")),
+                true);
+
+            string netCoreAppPathDepsJson = Path.Combine(netCoreAppPath, "Microsoft.NETCore.App.deps.json");
+
+            string depsJsonBody = $@"{{
+              ""runtimeTarget"": "".NETCoreApp"",
+              ""targets"": {{
+                "".NETCoreApp"": {{
+                  ""Microsoft.NETCore.App/{version}"": {{
+                    ""native"": {{
+                      ""runtimes/any/native/libcoreclr.so"": {{ }}
```

Comment:
Use `RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform`

Reply:
I think it just needs the library name, not a relative path?

Reply:
> Use RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform

Good catch.  

> I think it just needs the library name, not a relative path?

I copied it from `preview5` & `2.1.7`.  They both had a path.  I left it because it is working.

Reply:
I just tried removing the "virtual" path.  It actually breaks the install so the runtime is not recognized.